File: modules/module_05/src/smaceps.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

float smaceps()
{
  float one = 1.0;
  float eps = 1.0;
  float appone = one + eps;

  float error = fabs(appone-one);
  printf("%g", error);
  while(error > 0.0)
  {
    eps = eps / 2.0;
    appone = one + eps;
    error = fabs( appone - one );
  }
}

int main()
{
  printf("error = %g\n", smaceps());
}

File: modules/module_05/src/smaceps.py
import numpy as np

eps = float(1.0)
one = float(1.0)
appone = float(one + eps)

error = float(np.abs(appone - one))

for i in range(1,40):
    print(i, error)
    eps = float(eps) / 10.
    appone = one + eps
    error = float(np.abs(appone - one))
    if error == 0.0:
        break
